src/addrman.cpp

```diff
@@ -962,6 +970,28 @@ std::optional<AddressPosition> AddrManImpl::FindAddressEntry_(const CAddress& ad
     }
 }
 
+size_t AddrManImpl::size_(std::optional<Network> net, std::optional<bool> in_new) const
+{
+    AssertLockHeld(cs);
+
+    if (!net.has_value()) {
+        if (!in_new.has_value()) {
+            return nNew + nTried;
+        } else {
+            return *in_new ? nNew : nTried;
+        }
+    }
+    if (auto it = m_map_counts.find(*net); it != m_map_counts.end()) {
+        auto entry = it->second;
+        if (!in_new.has_value()) {
+            return entry.first + entry.second;
+        } else {
+            return *in_new ? entry.first : entry.second;
+        }
+    }
```

Comment:
Yes, these need to be identical, but I actually changed it to use the old `vRandom.size()` used in `size()` so that isn't a change from the status quo when the two `size()` functions are merged as you suggest.